controllers/organisation.go

```diff
@@ -48,17 +48,137 @@ func (c *NewController) CreateOrganisation(w http.ResponseWriter, r *http.Reques
 		json.NewEncoder(w).Encode(apiResponse.Error(utility.BAD_REQUEST, errMsg))
 		return
 	}
+	val := c.Repo.AlreadyOrganisation(requestData)
+	if val == true {
+		log.Error("Organisation name Already Exist")
+		json.NewEncoder(w).Encode(apiResponse.Error(utility.ALREADY_EXIST, utility.GetCodeMsg(utility.ALREADY_EXIST)))
+		return
+	}
 
 	// Pass request to repo
-    response ,err = c.Repo.CreateOrganisation(requestData)
-      if err != nil {
-		  log.Error("Error occurred while creating organisation, %s", err.Error())
-		  w.WriteHeader(http.StatusOK)
-		  json.NewEncoder(w).Encode(apiResponse.Error(utility.SERVER_ERROR, utility.GetCodeMsg(utility.SERVER_ERROR)))
-	 	  return
-      }
-
-	  log.Info("Response body: %+v", response)
-	  w.WriteHeader(http.StatusOK)
-      json.NewEncoder(w).Encode(apiResponse.Success(utility.SYSTEM001, utility.GetCodeMsg(utility.SYSTEM001), response))
+	response, err = c.Repo.CreateOrganisation(requestData)
+	if err != nil {
+		log.Error("Error occurred while creating organisation, %s", err.Error())
+		w.WriteHeader(http.StatusOK)
+		json.NewEncoder(w).Encode(apiResponse.Error(utility.SERVER_ERROR, utility.GetCodeMsg(utility.SERVER_ERROR)))
+		return
+	}
+
+	log.Info("Response body: %+v", response)
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(apiResponse.Success(utility.SYSTEM001, utility.GetCodeMsg(utility.SYSTEM001), response))
+}
+
+func (c *NewController) CreateBranch(w http.ResponseWriter, r *http.Request) {
+	log := c.Logger.NewContext()
+	log.SetLogID(r.Header.Get("LogID"))
+	log.Info("creating branch")
+	response := utility.NewResponse()
+	var br *dto.CreateBranch
+	err := json.NewDecoder(r.Body).Decode(&br)
+	if err != nil {
+		log.Error("Error occured while parsing the request body, %s", err.Error())
+		json.NewEncoder(w).Encode(response.Error(utility.BAD_REQUEST, utility.GetCodeMsg(utility.BAD_REQUEST)))
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	log.Debug("Request body: %+v", &br)
+
+	validationRes := validator.New()
+	if err := validationRes.Struct(&br); err != nil {
+		validationErrors := err.(validator.ValidationErrors)
+		c.Logger.Error("Validation failed on some fields : %+v", validationErrors)
+		w.WriteHeader(http.StatusOK)
+		json.NewEncoder(w).Encode(response.ValidationError(utility.VALIDATIONERR, utility.GetCodeMsg(utility.VALIDATIONERR), validationErrors.Error()))
+		return
+	}
+	val := c.Repo.AlreadyBranch(br)
```

Comment:
"AlreadyBranch" is not a good name for the function try "DoesBranchExist" or "BranchExist".